src/main/java/com/intel/genomicsdb/GenomicsDBImporter.java

```diff
@@ -23,1615 +23,1664 @@
 package com.intel.genomicsdb;
 
 import com.googlecode.protobuf.format.JsonFormat;
+import com.intel.genomicsdb.model.BaseImportConfig;
 import htsjdk.samtools.util.CloseableIterator;
 import htsjdk.samtools.util.RuntimeIOException;
 import htsjdk.tribble.AbstractFeatureReader;
 import htsjdk.tribble.FeatureReader;
 import htsjdk.variant.variantcontext.VariantContext;
-import htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder;
 import htsjdk.variant.variantcontext.writer.VariantContextWriter;
-import htsjdk.variant.vcf.VCFHeaderLine;
-import htsjdk.variant.vcf.VCFHeader;
-import htsjdk.variant.vcf.VCFContigHeaderLine;
-import htsjdk.variant.vcf.VCFFilterHeaderLine;
-import htsjdk.variant.vcf.VCFFormatHeaderLine;
-import htsjdk.variant.vcf.VCFInfoHeaderLine;
-import htsjdk.variant.vcf.VCFConstants;
-import htsjdk.variant.vcf.VCFHeaderLineType;
-import htsjdk.variant.vcf.VCFHeaderLineCount;
-import org.apache.commons.io.FileUtils;
+import htsjdk.variant.variantcontext.writer.VariantContextWriterBuilder;
+import htsjdk.variant.vcf.*;
 import org.json.simple.JSONArray;
 import org.json.simple.JSONObject;
 import org.json.simple.parser.JSONParser;
 import org.json.simple.parser.ParseException;
-import java.io.IOException;
-import java.io.FileNotFoundException;
-import java.io.File;
-import java.io.PrintWriter;
-import java.io.StringWriter;
+
+import java.io.*;
 import java.util.*;
+import java.util.concurrent.*;
+import java.util.stream.Collectors;
 
 import static com.googlecode.protobuf.format.JsonFormat.printToString;
 
 /**
  * Java wrapper for vcf2tiledb - imports VCFs into TileDB/GenomicsDB.
  * All vid information is assumed to be set correctly by the user (JSON files)
  */
-public class GenomicsDBImporter
-{
-  static
-  {
-    try
-    {
-      boolean loaded = GenomicsDBUtils.loadLibrary();
-      if(!loaded)
-        throw new GenomicsDBException("Could not load genomicsdb native library");
-    }
-    catch(UnsatisfiedLinkError ule)
-    {
-      throw new GenomicsDBException("Could not load genomicsdb native library", ule);
+public class GenomicsDBImporter {
+    private static final String DEFAULT_ARRAY_NAME = "genomicsdb_array";
+    private static final int DEFAULT_TILEDB_CELLS_PER_TILE = 1000;
+    private static final int DEFAULT_ZERO_BATCH_SIZE = 0;
+    //Allele specific annotation fields
+    private static final HashSet<String> mRLengthHistogramFieldsWithFloatBins = new HashSet<>(Arrays.asList(
+            "AS_RAW_BaseQRankSum",
+            "AS_RAW_MQRankSum",
+            "AS_RAW_ReadPosRankSum"
+    ));
+    private static final HashSet<String> mRLengthTwoDFloatVectorFields = new HashSet<>(Collections.singletonList(
+            "AS_RAW_MQ"
+    ));
+    private static final HashSet<String> mRLengthTwoDIntVectorFields = new HashSet<>(Collections.singletonList(
+            "AS_SB_TABLE"
+    ));
+    static final String CHROMOSOME_INTERVAL_FOLDER = "%s_%d_%d";
```

Comment:
Checked with the new reference genome - the chromosome names have "\_". No '#' or '$' - so those chars should be fine :(